src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -211,6 +215,28 @@ public void transform(KeyValue<LongWritable, GenericData.Record> input, Emitter<
 
   @Override
   public void onRunFinish(boolean succeeded, BatchSourceContext context) {
+    deleteGcsTemporaryDirectory();
+    deleteBigQueryTemporaryTable();
+  }
+
+  private void deleteBigQueryTemporaryTable() {
+    String temporaryTable = configuration.get(BigQueryConstants.CONFIG_TEMPORARY_TABLE_NAME);
+    if (temporaryTable != null) {
+      String serviceAccount = config.getServiceAccount();
+      Credentials credentials = null;
+      try {
+        credentials = serviceAccount == null ?
+                null : GCPUtils.loadServiceAccountCredentials(serviceAccount, config.isServiceAccountFilePath());
+      } catch (IOException e) {
+        LOG.warn("Failed to load service account credentials: {}", e.getMessage());
```

Comment:
if service account is failed to load is it safe to proceed to delete the table?

Reply:
Good catch. If the service account fails we are not proceeding delete table.

Reply:
Updated